### PR TITLE
fix(appCreation): Don't send error message on app creation if no error

### PR DESCRIPTION
**Pull request by karlomendozaarmory**

### CODE REVIEW

**Comment by karlomendozaarmory** [comment]
This error was from 
https://armory.atlassian.net/browse/SUPPORT-801

Fix is to remove the 404 error message when the app creation works

**Comment by akshayabd** [comment]
Thanks for working to fix this - its an annoying bug - I will test once this is merged

**Comment by dmrogers7** [comment]
Why hasn't this been merged?  This is an annoying issue that causes confusion because the 404 error condition reported because the application does not exist is handled by the code.  

More importantly though, because the 404 error is returned by the `updatePipelines` function, it causes [ProcessDinghyFile](https://github.com/armory/dinghy/blob/16269a7cd9c2c8d6eb29fdc096efbc3cba7ce806/pkg/dinghyfile/builder.go#L219) to return an error, which in turns causes all subsequent dinghy file processing to be aborted by breaking out of the loops in [RebuildModuleRoots](https://github.com/armory/dinghy/blob/16269a7cd9c2c8d6eb29fdc096efbc3cba7ce806/pkg/dinghyfile/builder.go#L267) and [ProcessPush](https://github.com/armory/dinghy/blob/16269a7cd9c2c8d6eb29fdc096efbc3cba7ce806/pkg/web/routes.go#L480)

**Comment by akshayabd** [comment]
@Jossuecito @dogonthehorizon could we get this reviewed and merged please?

File: pkg/dinghyfile/builder.go [hunk 1]
```diff
@@ -251,9 +251,11 @@ func (b *PipelineBuilder) updatePipelines(app *plank.Application, pipelines []pl
 		if errunsup.Code == 404 {
 			// Likely just not there...
 			b.Logger.Infof("Creating application '%s'...", app.Name)
-			if err := b.Client.CreateApplication(app); err != nil {
+			if err = b.Client.CreateApplication(app); err != nil {
 				b.Logger.Errorf("Failed to create application (%s)", err.Error())
 				return err
+			} else {
+				err = nil
```
Comment (Jossuecito) [inline pkg/dinghyfile/builder.go:258]: I think this else can be deleted? Since err should be nil at that moment

File: pkg/dinghyfile/builder.go [hunk 2]
```diff
@@ -251,9 +251,11 @@ func (b *PipelineBuilder) updatePipelines(app *plank.Application, pipelines []pl
 		if errunsup.Code == 404 {
 			// Likely just not there...
 			b.Logger.Infof("Creating application '%s'...", app.Name)
-			if err := b.Client.CreateApplication(app); err != nil {
+			if err = b.Client.CreateApplication(app); err != nil {
```
Comment (Jossuecito) [inline pkg/dinghyfile/builder.go:254]: I think changing just the error variable name would be easier, also for this you can delete the else part?

**Review by dogonthehorizon — APPROVED** [review]

**Pull request merged**